### PR TITLE
Support address literals

### PR DESCRIPTION
**Pull request by DJRHails**
Addresses #233

### CODE REVIEW

File: Sources/AST/Token.swift [hunk 1]
```diff
@@ -115,6 +115,7 @@ extension Token {
       case boolean(BooleanLiteral)
       case decimal(DecimalLiteral)
       case string(String)
+      case address(Int)
```
Comment (bearbin) [inline Sources/AST/Token.swift]: A Swift Int is not going to be big enough to hold a 160 bit address, is it? I think a larger data type should go here, and add a test to make sure the whole range of addresses is covered.

Comment (DJRHails) [reply Sources/AST/Token.swift]: Almost certainly true. Did not think about that.

File: Sources/Parser/Tokenizer.swift [hunk 2]
```diff
@@ -52,6 +52,12 @@ public struct Tokenizer {
         } else {
           tokens.append(Token(kind: .literal(.decimal(.integer(num))), sourceLocation: sourceLocation))
         }
+      } else if component.hasPrefix("0x"){
+        // The token is an address literal.
+        let hex = component.dropFirst(2)
```
Comment (franklinsch) [inline Sources/Parser/Tokenizer.swift]: I don't think we need to drop the leading "0x", except if you have a good reason

File: Sources/Parser/Tokenizer.swift [hunk 3]
```diff
@@ -52,6 +52,12 @@ public struct Tokenizer {
...
+        if(hex.count == 40){
```
Comment (franklinsch) [inline Sources/Parser/Tokenizer.swift]: Spacing

File: Tests/ParserTests/literals.flint [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+// CHECK-AST: BinaryExpression
+// CHECK-AST:  VariableDeclaration
+// CHECK-AST:    let
+// CHECK-AST:    identifier "a"
+// CHECK-AST:    built-in type Address
+// CHECK-AST:  =
+// CHECK-AST:    literal 0x0
+func zoo() -> Address {
+let a: Address = 0x0000000000000000000000000000000000000000
```
Comment (franklinsch) [inline Tests/ParserTests/literals.flint]: Indentation

File: Sources/Parser/Tokenizer.swift (hunk 2 above)
Comment (DJRHails) [reply Sources/Parser/Tokenizer.swift]: Mainly related to the Int conversion - without it there is rightly no need

**Comment by franklinsch** [comment]
We need to check whether address literals have the appropriate length in the Semantic Analyzer, once #241 is merged.

**Pull request merged**